lib/p2p/NodeList.ts

```diff
@@ -95,12 +95,17 @@ class NodeList extends EventEmitter {
   }
 
   /**
-   * Create a Node in the database.
+   * Creates a Node in the database if it does not exist.
    */
-  public createNode = async (nodeFactory: NodeFactory): Promise<NodeInstance> => {
-    const node = await this.repository.addNode(nodeFactory);
-    this.nodes.set(node.nodePubKey, node);
-    return node;
+  public createNode = async (nodeFactory: NodeFactory) => {
+    try {
+      const node = await this.repository.addNode(nodeFactory);
+      this.nodes.set(node.nodePubKey, node);
+    } catch (err) {
+      if (err.name !== 'SequelizeUniqueConstraintError') {
```

Comment:
It feels weird that we're expecting this to occasionally error :thinking:. Don't have a better solution/recommendation - just pointing this out.

Reply:
You're right it is weird, I think it really only comes up in tests where p2p latency is ~zero. But the end result is that we have the node in the database and in the nodelist which is the desired state, and catching the uniqueness error is simpler than tracking a list of pending nodes that are currently being written to the db. I will explain here with a comment.

Reply:
Actually, I'm going to go ahead and move this code relating to sequelize and the database over to the `P2PRepository` class since that's a more appropriate place for it.